Recipes without descriptions can't be updated

You get a Dynamo update expression if you try to update a recipe that does not have a description.

```
Caused by: com.amazonaws.services.dynamodbv2.model.AmazonDynamoDBException:
Invalid UpdateExpression: An expression attribute value used in expression is not defined; 
attribute value: :l_l_l_l_l_update (Service: AmazonDynamoDBv2; Status Code: 400; Error Code: ValidationException; Request ID: 99ISP9BG3VA2D46N2VPV099UUVVV4KQNSO5AEMVJF66Q9ASUAAJG)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1579)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1249)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1030)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:742)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:716)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:699)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:667)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:649)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:513)
	at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.doInvoke(AmazonDynamoDBClient.java:1974)
	at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.invoke(AmazonDynamoDBClient.java:1950)
	at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.updateItem(AmazonDynamoDBClient.java:1708)
	at com.gu.scanamo.ops.ScanamoInterpreters$$anon$1.apply(ScanamoInterpreters.scala:75)
	at com.gu.scanamo.ops.ScanamoInterpreters$$anon$1.apply(ScanamoInterpreters.scala:50)
	at cats.free.Free$$anonfun$foldMap$1.apply(Free.scala:126)
	at cats.free.Free$$anonfun$foldMap$1.apply(Free.scala:124)
	at cats.package$$anon$1.tailRecM(package.scala:36)
	at cats.free.Free.foldMap(Free.scala:124)
	at cats.free.Free$$anonfun$foldMap$1.apply(Free.scala:127)
	at cats.free.Free$$anonfun$foldMap$1.apply(Free.scala:124)
	at cats.package$$anon$1.tailRecM(package.scala:36)
	at cats.free.Free.foldMap(Free.scala:124)
	at com.gu.scanamo.Scanamo$.exec(Scanamo.scala:17)
	at data.Dynamo$RichScanamoOps$.exec$extension(Dynamo.scala:97)
	at data.Recipes$.update(Recipes.scala:55)
	at controllers.RecipeController$$anonfun$updateRecipe$1$$anonfun$apply$6$$anonfun$apply$8.apply(RecipeController.scala:59)
	at controllers.RecipeController$$anonfun$updateRecipe$1$$anonfun$apply$6$$anonfun$apply$8.apply(RecipeController.scala:54)

...
```

The update is defined as below:

```scala
val update = table.update('id -> recipe.id,
      set('description -> description) and
        set('baseImageId -> baseImage.id) and
        set('roles -> roles) and
        set('modifiedBy -> modifiedBy) and
        set('modifiedAt -> DateTime.now()) and
        (if (bakeSchedule.isDefined) set('bakeSchedule -> bakeSchedule) else remove('bakeSchedule))
    )
```

I think the fix would be to only include `Description` in the update expression if it has been set, like we do for `bakeSchedule`?

Probably introduced by #28

--- a/app/data/Recipes.scala
+++ b/app/data/Recipes.scala
@@ -47,15 +47,19 @@ object Recipes {
     modifiedBy: String,
     bakeSchedule: Option[BakeSchedule])(implicit dynamo: Dynamo): Either[DynamoReadError, Recipe] = {
 
-    val update = table.update('id -> recipe.id,
-      set('description -> description) and
-        set('baseImageId -> baseImage.id) and
+    val baseUpdateExpr =
+      set('baseImageId -> baseImage.id) and
         set('roles -> roles) and
         set('modifiedBy -> modifiedBy) and
         set('modifiedAt -> DateTime.now()) and
         (if (bakeSchedule.isDefined) set('bakeSchedule -> bakeSchedule) else remove('bakeSchedule))
-    )
 
+    val updateExpr = description match {
+      case Some(desc) => baseUpdateExpr and set('description -> desc)
+      case None => baseUpdateExpr
+    }
+
+    val update = table.update('id -> recipe.id, updateExpr)
     update.exec().map(Recipe.db2domain(_, baseImage))
   }
 

